deprecate RBD plugin from available in-tree drivers

humblec opened the pull request:
Based on
https://groups.google.com/g/kubernetes-sig-storage/c/h5751_B5LQM, the consensus was to start the deprecation in v1.28.

This commit start the deprecation process of RBD plugin from in-tree drivers.


/kind deprecation


```release-note
ACTION REQUIRED:
RBD volume plugin ( `kubernetes.io/rbd`) has been deprecated in this release and will be removed in a subsequent release. Alternative is to use RBD CSI driver (https://github.com/ceph/ceph-csi/) in your Kubernetes Cluster.
```


```docs

```

carlory commented:
Sorry, It‘s my mistake. Let's continue with this pr.

carlory commented on pkg/api/persistentvolume/util.go:
```diff
@@ -95,6 +95,8 @@ func warningsForPersistentVolumeSpecAndMeta(fieldPath *field.Path, pvSpec *api.P
 		warnings = append(warnings, fmt.Sprintf("%s: deprecated in v1.25, non-functional in v1.26+", fieldPath.Child("spec", "persistentVolumeSource").Child("glusterfs")))
 	}
-
+	if pvSpec.RBD != nil {
+		warnings = append(warnings, fmt.Sprintf("%s: deprecated in v1.28, non-functional in v1.31+", fieldPath.Child("spec", "persistentVolumeSource").Child("rbd")))
+	}
```
```suggestion
	if pvSpec.RBD != nil {
		warnings = append(warnings, fmt.Sprintf("%s: deprecated in v1.28, non-functional in v1.31+", fieldPath.Child("spec", "rbd"))
	}
```

carlory commented on pkg/api/persistentvolume/util_test.go:
```diff
@@ -212,6 +212,27 @@ func TestWarnings(t *testing.T) {
+				},
+			},
+			expected: []string{
+				`spec.persistentVolumeSource.rbd: deprecated in v1.28, non-functional in v1.31+`},
```
```suggestion
				`spec.rbd: deprecated in v1.28, non-functional in v1.31+`},
```

carlory reviewed (commented):
the `persistentVolumeSource ` field is invisible to the user. 

```
[root@master ~]# kubectl explain pv.spec.persistentVolumeSource.rbd
KIND:       PersistentVolume
VERSION:    v1

error: field "rbd" does not exist

[root@master ~]# kubectl explain pv.spec.rbd
KIND:       PersistentVolume
VERSION:    v1

FIELD: rbd <RBDPersistentVolumeSource>
```

so we need remove it.

carlory commented:
ditto

https://github.com/kubernetes/kubernetes/blob/f956675e2a3dbc52b65c0189b4c19d257fb62596/pkg/api/persistentvolume/util.go#L83-L97

humblec committed:
deprecate RBD plugin from available in-tree drivers

Based on https://groups.google.com/g/kubernetes-sig-storage/c/h5751_B5LQM, the
consensus was to start the deprecation in v1.28.

This commit start the deprecation process of RBD plugin from in-tree
drivers.

ACTION REQUIRED:
   RBD volume plugin ( `kubernetes.io/rbd`) has been deprecated in this release
   and will be removed in a subsequent release. Alternative is to use RBD CSI driver
   (https://github.com/ceph/ceph-csi/) in your Kubernetes Cluster.

Signed-off-by: Humble Chirammal <humble.devassy@gmail.com>

humblec committed:
Change the fieldpath in the deprecated plugin response

Signed-off-by: Humble Chirammal <humble.devassy@gmail.com>
Signed-off-by: carlory <baofa.fan@daocloud.io>

jsafrane commented:
/lgtm
/label api-review
